### PR TITLE
Dev/movie rating

### PR DESCRIPTION
**Pull request by quappin**
Added implementation for movie rating. The user can give a movie of their choosing a 1-5 star rating and the program saves this rating to the database.

### CODE REVIEW

File: movie_funcs.py [hunk 1]
```diff
@@ -244,3 +244,32 @@ def browse_movies(conn) -> int:
...
+
+        if existing_rating:
+            # If the user has already rated this movie, update their rating
+            curs.execute("UPDATE rated SET rating = %s WHERE userid = %s AND movieid = %s", (rating, user_id, movie_id))
+            print(f"Updated your rating for this movie to {rating} stars!")
+        else:
+            # Otherwise, assign the movie a new rating
+            curs.execute("INSERT INTO rated (userid, movieid, rating) VALUES (%s, %s, %s)", (user_id, movie_id, rating))
+            print(f"You gave this movie a {rating} star rating!")
+
```
Comment (armedturret) [inline movie_funcs.py:274]: Need to call conn.commit() here for changes to apply.

**Review by armedturret — CHANGES_REQUESTED** [review]
Looks good! Only found one bug (caused by a crash in my code (whoops)

File: movie_funcs.py [hunk 2]
```diff
@@ -244,3 +244,32 @@ def browse_movies(conn) -> int:
...
+def rate_movie(conn, user_id, movie_id):
+    """
+    Assigns a movie a (1-5 star) rating given by a user.
+
+    :param conn: Connection to the database.
+    :param user_id: The ID of the user rating a movie.
+    :param movie_id: The ID of the movie being rated.
+    """
+
+    rating = int(input_utils.get_input_matching("Enter a rating (1-5): ", regex="^[1-5]$"))
```
Comment (armedturret) [inline movie_funcs.py]: We should be able to give movies a 0 star rating

File: sigmadb.py [hunk 3]
```diff
@@ -95,7 +95,16 @@ def main():
 
                     match action:
                         case "2":
-                            movie_funcs.browse_movies(conn)
+                            selected_movie_id = movie_funcs.browse_movies(conn)
+                            if selected_movie_id == -1:
+                                break
```
Comment (armedturret) [inline sigmadb.py]: This exists the application (maybe use pass?)

File: sigmadb.py [hunk 4]
```diff
@@ -95,7 +95,16 @@ def main():
...
+                            else:
+                                watch_or_rate = input_utils.get_input_matching("1 - watch movie\n2 - rate movie\n> ", regex="[12]")
+                                # add functionality for view_movie in later branch
+                                if watch_or_rate == "2":
+                                    movie_funcs.rate_movie(conn, userid, selected_movie_id)
+                                else:
+                                    break
```
Comment (armedturret) [inline sigmadb.py]: Ditto here. Pass might be more appropriate.

**Review by armedturret — CHANGES_REQUESTED** [review]

**Review by quappin — COMMENTED** [review]
I made the requested changes and pushed them, this should be good to go now

**Review by armedturret — APPROVED** [review]
LGTM

**Pull request merged**